src/core/include/openvino/runtime/tensor_util.hpp

```diff
@@ -0,0 +1,89 @@
+///              One of the dimensions can be dynamic. In this case it will be determined automatically based on the
+///              length of the file content and `offset`. Default value is [?].
+/// \param offset Read file starting from specified offset. Default is 0. The remining size of the file should be
+/// compatible with shape.
+/// \param mmap Use mmap that postpones real read from file until data is accessed. If mmap is used, the file
+///             should not be modified until returned tensor is destroyed.
+OPENVINO_API
+Tensor read_tensor_data(const std::string& file_name,
+                        const element::Type& element_type = element::u8,
+                        const PartialShape& shape = PartialShape{Dimension::dynamic()},
+                        std::size_t offset = 0,
+                        bool mmap = true);
+
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)
+OPENVINO_API
+Tensor read_tensor_data(const std::wstring& file_name,
+                        const element::Type& element_type = element::u8,
+                        const PartialShape& shape = PartialShape{Dimension::dynamic()},
+                        std::size_t offset = 0,
+                        bool mmap = true);
+#endif
+OPENVINO_API
+Tensor read_tensor_data(const std::filesystem::path& file_name,
+                        const element::Type& element_type = element::u8,
+                        const PartialShape& shape = PartialShape{Dimension::dynamic()},
+                        std::size_t offset = 0,
+                        bool mmap = true);
+
+/// \brief Read raw data from a file into pre-allocated tensor.
+/// \param file_name Path to the input file with raw tensor data.
+/// \param tensor Tensor to read data to. Tensor should have correct element_type and shape set that is used to
+/// determine how many bytes will be read from the file.
+/// \param offset Read file starting from specified offset. Default
+/// is 0. The remining part of the file should contain enough bytes to satisfy tensor size.
+OPENVINO_API
+void read_tensor_data(const std::string& file_name, Tensor& tensor, std::size_t offset = 0);
+#if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT)
+OPENVINO_API
+void read_tensor_data(const std::wstring& file_name, Tensor& tensor, std::size_t offset = 0);
+#endif
+OPENVINO_API
+void read_tensor_data(const std::filesystem::path& file_name, Tensor& tensor, std::size_t offset = 0);
+
+/// \brief Save given tensor data into a temporary file. Read the content from the file to a new tensor using mmap.
+///        The temporary file is removed when the returned tensor and all its copies are destroyed.
+/// \param tensor Tensor to read data to. Tensor should have correct element_type and shape set that is used to
+/// \param file_name Path to the temporary file
+
+OPENVINO_API
+Tensor create_mmaped_tensor(const Tensor& tensor, const std::string& file_name);
```

Comment:
Maybe these functions and some read function should be new ctor to create tensor from data from file (using mmap or not)?
Also tensor could have read/save members to load/store data from/to file.

Reply:
According to @slyalin we separate (de)serialization from classes like it made for model.
Just a logical separation, no technical basis